src/services/factories/tokenList.ts

```diff
@@ -133,7 +133,7 @@ export function getTokensFactory(
   }
 
   async function fetchTcrAddresses(networkId: number): Promise<Set<string>> {
-    return tcrApi ? new Set(await tcrApi.getTokens(networkId)) : new Set()
+    return new Set(tcrApi && (await tcrApi.getTokens(networkId)))
```

Comment:
What does `new Set<string>(false)` return?

Reply:
`new Set(false)` breaks
but here `tcrApi` is `TcrApi | undefined`
`new Set(undefined)` is same as `new Set()`

Reply:
ohhhh

Reply:
I had the same first impression as Leandro, then I realised
I have to admit I'm not super fan of relying on this logics, I prefer to be explicit, even on the cost of being slightly more ineficient. But in this case, the solution was good, was not that complicated, and makes dima happier

Reply:
It's ok to push harder for your way. I don't much care as long as it works
Ternary is ternary :+1: